std/typecons.d

```diff
@@ -792,10 +852,36 @@ template Tuple(Specs...)
          * Returns:
          *     The string representation of this `Tuple`.
          */
-        string toString()()
+        string toString()() const
         {
-            import std.conv : to;
-            return this.to!string;
+            import std.array : appender;
+            auto app = appender!string();
+            this.toString((const(char)[] chunk) => app ~= chunk);
+            return app.data;
+        }
+
+        ///
+        unittest
+        {
+            import std.format    : format, FormatException;
+            import std.format : format;
```

Comment:
I think it's definitely good to test, but I'm not sure whether it's worth including in the *documented* unit test.

However, `Tuple`'s unit tests might be or become a performance issue; since they are templated, there are as many copies of the unittest as there are unique template instantiations. In `std.container` we reverted to using DDoc example sections and moving the unittests outside the template because of this issue, despite the source code duplication it causes. That sort of change is beyond the scope of this PR, but I thought I might mention it anyway.